pkg/resource/deploy/step_generator.go

```diff
@@ -440,7 +464,7 @@ func (sg *stepGenerator) generateSteps(event RegisterResourceEvent) ([]Step, res
 
 		// If the user requested only specific resources to update, and this resource was not in
 		// that set, then do nothin but create a SameStep for it.
-		if !sg.isTargetedForUpdate(urn) {
+		if !sg.isTargetedForUpdateOrAddIfDep(urn, goal.Dependencies) {
```

Comment:
Wondering what's goal.Dependencies here in the context, immediate or transitive dependencies of resource identified by urn, or something else? How does it relate to ResourceGraph?

Also I have a premonition that the engine on the update path observes the dependency graph incrementally; does it observe enough of it to do a good job here? Probably yes but good to think through.

Reply:
I'm pretty sure that the dependency graph is observed incrementally, but in topologically sorted order. At least, that is the assumption I am working under. It's fine if `goal.Dependencies` is local (not transitive). Transitive parents will already have been registered in previous calls.

Reply:
> I'm pretty sure that the dependency graph is observed incrementally, but in topologically sorted order

This is correct.

> It's fine if goal.Dependencies is local (not transitive). Transitive parents will already have been registered in previous calls.

`goal.Dependencies` is indeed non-transitive. Anything that a resource R depends on--whether the edge is a dependency edge, a parent edge, or a provider edge--must be registered prior to R itself.